File: workspace/src/lib/process_data.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const inputPath = path.join(__dirname, '../data/raw_data.tsv');
const outputPath = path.join(__dirname, '../data/products.json');

// Helper to convert " R$ 6,60 " -> 6.60
function parsePrice(priceStr) {
  if (!priceStr) return 0;
  let cleanStr = priceStr.replace('R$', '').trim().replace('.', '').replace(',', '.');
  return parseFloat(cleanStr) || 0;
}

try {
  let content;
  try {
    content = fs.readFileSync(inputPath, 'utf8');
  } catch (err) {
    if (err.code === 'ENOENT') {
      console.error(`ERROR: Could not find raw file at ${inputPath}`);
      console.error("Please save your spreadsheet data to 'src/data/raw_data.tsv' and try again.");
      process.exit(1);
    }
    throw err;
  }

  const lines = content.split('\n').filter(line => line.trim() !== '');

  const products = lines.map(line => {
    const cols = line.split('\t');
    
    // We expect 8 to 9 columns based on the sample
    const locationCode = cols[0]?.trim() || '';
    const internalId = cols[1]?.trim() || '';
    const category = cols[2]?.trim() || '';
    const subcategory = cols[3]?.trim() || '';
    const description = cols[4]?.trim() || '';
    const ncm = cols[5]?.trim() || '';
    const sku = cols[6]?.trim() || '';
    const priceStr = cols[7]?.trim() || '';
    const expiryDate = cols[8]?.trim() || '';

    const price = parsePrice(priceStr);

    return {
      id: internalId ? `PROD-${internalId}` : locationCode,
      name: sku ? `${sku} - ${description}` : description,
      description: `Categoria: ${category} | Linha: ${subcategory} | Código: ${locationCode} | NCM: ${ncm}. ${expiryDate ? 'Validade: ' + expiryDate : ''}`,
      price: price,
      unit: 'unidade',
      stock: 10,  // Default stock since it is not provided
      category: category,
      image: '/images/filter.png', // Temporary placeholder
      ncm: ncm,
      location: locationCode,
      sku: sku
    };
  });

  fs.writeFileSync(outputPath, JSON.stringify(products, null, 2), 'utf8');
  console.log(`Successfully processed ${products.length} products and saved to products.json`);

} catch (e) {
  console.error('Error processing data:', e);
}

File: workspace/src/lib/process_data.py
import json
import os
import sys

input_path = os.path.join(os.path.dirname(__file__), '..', 'data', 'raw_data.tsv')
output_path = os.path.join(os.path.dirname(__file__), '..', 'data', 'products.json')

def parse_price(price_str):
    if not price_str:
        return 0.0
    clean_str = price_str.replace('R$', '').strip().replace('.', '').replace(',', '.')
    try:
        return float(clean_str)
    except ValueError:
        return 0.0

try:
    with open(input_path, 'r', encoding='utf-8') as f:
        lines = [line.strip() for line in f if line.strip()]
except FileNotFoundError:
    print(f"ERROR: Could not find raw file at {input_path}")
    sys.exit(1)

products = []

for line in lines:
    cols = line.split('\t')
    location_code = cols[0].strip() if len(cols) > 0 else ''
    internal_id = cols[1].strip() if len(cols) > 1 else ''
    category = cols[2].strip() if len(cols) > 2 else ''
    subcategory = cols[3].strip() if len(cols) > 3 else ''
    description = cols[4].strip() if len(cols) > 4 else ''
    ncm = cols[5].strip() if len(cols) > 5 else ''
    sku = cols[6].strip() if len(cols) > 6 else ''
    price_str = cols[7].strip() if len(cols) > 7 else ''
    expiry_date = cols[8].strip() if len(cols) > 8 else ''

    price = parse_price(price_str)
    
    prod_id = f"PROD-{internal_id}" if internal_id else location_code
    name = f"{sku} - {description}" if sku else description

    desc_details = f"Categoria: {category} | Linha: {subcategory} | Código: {location_code} | NCM: {ncm}."
    if expiry_date:
        desc_details += f" Validade: {expiry_date}"

    products.append({
        "id": prod_id,
        "name": name,
        "description": desc_details,
        "price": price,
        "unit": "unidade",
        "stock": 10,
        "category": category,
        "image": "/images/filter.png",
        "ncm": ncm,
        "location": location_code,
        "sku": sku
    })

with open(output_path, 'w', encoding='utf-8') as f:
    json.dump(products, f, indent=2, ensure_ascii=False)

print(f"Successfully processed {len(products)} products and saved to products.json")
